### PR TITLE
#23 : RevokeWorkRequestController/Save the new/modified entities

### PR DESCRIPTION
**Pull request by rameshthumati22**

### CODE REVIEW

File: src/main/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestController.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.kodekonveyor.work_request.WorkRequestRepository;
+
+@Controller
+public class RevokeWorkRequestController {
+
+  @Autowired
+  private WorkRequestRepository workRequestRepository;
+
+  public void call(final String workRequestId) {
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestController.java]: Please add a @RequestMapping annotation with the method and path.

**Comment by magwas** [comment]
please note that I have updated this branch with the latest develop. it is newer than what you have on disk.

File: src/main/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestController.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    workRequestRepository.deleteById(Long.parseLong(workRequestId));
```
Comment (szirbucz) [inline src/main/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestController.java]: workRequestId should be long. It makes no sense to get it as String and convert it to long.

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.kodekonveyor.annotations.TestedBehaviour;
+import com.kodekonveyor.annotations.TestedService;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@RunWith(MockitoJUnitRunner.class)
+@TestedBehaviour("Save the new/modified entities")
+@TestedService("RevokeWorkRequestController")
+public class RevokeWorkRequestControllerTest
```
Comment (szirbucz) [inline src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java]: Negative test cases are missing.

**Comment by magwas** [comment]
New commit in develop, please resolve conflicts. If you have problems, just call me.

**Comment by magwas** [comment]
Somehow you have deleted all the TestData units, and you have conflicts with the latest develop.
If you need help call me.

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java (hunk 3 above)
Comment (szirbucz) [reply src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java]: I think we should throw an exception when no element has been deleted (the workRequestRepository.deleteById() method returns 0).
@magwas what is your opinion. Shall we throw an error or do nothing quietly when the id of the deleting element not found?

Comment (magwas) [reply src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java]: I think throwing a ValidationEception (which is handled by RestResponseEntityExceptionHandler ) is the way to go.

**Comment by magwas** [comment]
You have a test failure.

File: sh.exe.stackdump [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Stack trace:
```
Comment (magwas) [inline sh.exe.stackdump]: please remove this file

File: .settings/org.eclipse.jdt.core.prefs [hunk 5]
```diff
@@ -1,15 +1,17 @@
 eclipse.preferences.version=1
```
Comment (magwas) [inline .settings/org.eclipse.jdt.core.prefs]: please revert the changes to this file.

**Comment by magwas** [comment]
The line endings were previously MSDOS, and now they are unix. So that particular problem was in some previous commits, which this commit have fixed.

**Comment by magwas** [comment]
I do not see change in production code.
That means that the new test is redundant. However the new test is in the good place with the right annotations, so probably the old one should be ditched. We will discuss this case internally and come out with a policy about it today.

File: src/main/java/com/kodekonveyor/work_request/AddressUtil.java [hunk 6]
```diff
@@ -1,14 +1,14 @@
-package com.kodekonveyor.work_request;
```
Comment (szirbucz) [inline src/main/java/com/kodekonveyor/work_request/AddressUtil.java]: Whitespace only changes. Please set up your git to convert line endings to LF (linux style line ending)
https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  public void deleteWorkRequestById() {
+
+    revokeWorkRequestController
+        .call(WorkRequestEntityTestData.WORK_REQUEST_ID);
+
+    Mockito.verify(workRequestRepository)
+        .delete(WorkRequestEntityTestData.get());
+
+  }
+}
```
Comment (szirbucz) [inline src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java]: So, I investigated that why your test is green without production code. Someone, who made the RevokeWorkRequestController's validation mistakenly tested and implemented the repository call as well, and we haven't realized it on the code review.
Your test case is good as it is.
Please modify the RevokeWorkRequestControllerValidationTest.
Move the controller call from the setUp() method to the 
```
  @Test
  @DisplayName("Work request entity is saved")
  public void test5()
```
method and remove the verification of repository call there.

**Comment by magwas** [comment]
Plase see the development slack channel regarding rules handling this case.
@szirbucz will shortly comment about what to do in this particular case.

**Comment by magwas** [comment]
you have merge conflicts

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Test
+  @DisplayName("The work request with the given id is deleted")
+  public void deleteWorkRequestById() {
+
+    revokeWorkRequestController
+        .call(WorkRequestEntityTestData.WORK_REQUEST_ID);
+
+    Mockito.verify(workRequestRepository)
+        .delete(WorkRequestEntityTestData.get());
```
Comment (szirbucz) [inline src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerTest.java]: Remove this 2 lines

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerValidationTest.java [hunk 9]
```diff
@@ -1,75 +1,57 @@
...
-
-@ExtendWith(MockitoExtension.class)
-@MockitoSettings(strictness = Strictness.LENIENT)
-@RunWith(MockitoJUnitRunner.class)
-@TestedBehaviour("input validation")
-@TestedService("revokeWorkRequestController")
-public class RevokeWorkRequestControllerValidationTest
-    extends RevokeWorkRequestControllerTestBase {
-
-  @Override
```
Comment (szirbucz) [inline src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerValidationTest.java]: Don't remove the setUp method

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerValidationTest.java [hunk 10]
```diff
@@ -1,75 +1,57 @@
...
-  @BeforeEach
-  void setUp() {
-    super.setUp();
-    revokeWorkRequestController
```
Comment (szirbucz) [inline src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerValidationTest.java]: Move this call to the test5 method

File: src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerValidationTest.java [hunk 11]
```diff
@@ -1,75 +1,57 @@
...
-        .call(WorkRequestEntityTestData.WORK_REQUEST_ID);
-  }
-
-  @Test
```
Comment (szirbucz) [inline src/test/java/com/kodekonveyor/work_request/revoke/RevokeWorkRequestControllerValidationTest.java]: Don't remove this method

**Comment by szirbucz** [comment]
Please resolve merge conflicts

**Comment by magwas** [comment]
updated with develop, please pull locally

**Comment by magwas** [comment]
Closing due to inactivity. If you would like to work on it further, just grab the task again and reopen it.
